cmd/controller-manager/main.go

```diff
@@ -144,10 +164,107 @@ func main() {
 	setupLog.Info("Got config for ITS", "name", itsName)
 	itsRestConfig = itsClientLimits.LimitConfig(itsRestConfig)
 
+	// Start controllers based on leader election setting
+	if enableLeaderElection {
+		setupLog.Info("Starting with leader election enabled")
+		startControllersWithLeaderElection(ctx, setupLog, wdsRestConfig, itsRestConfig, wdsName, itsName, allowedGroupsSet, ctlrsToStart, wdsClientMetrics, itsClientMetrics)
+	} else {
+		setupLog.Info("Starting without leader election")
+		startControllersDirectly(ctx, setupLog, wdsRestConfig, itsRestConfig, wdsName, itsName, allowedGroupsSet, ctlrsToStart, wdsClientMetrics, itsClientMetrics)
+	}
+
+	// Start /readyz endpoint
+	go func() {
+		http.HandleFunc("/readyz", func(w http.ResponseWriter, r *http.Request) {
+			if readyFlag.Load() {
+				w.WriteHeader(http.StatusOK)
+				w.Write([]byte("ok"))
+			} else {
+				w.WriteHeader(http.StatusServiceUnavailable)
+				w.Write([]byte("not ready"))
+			}
+		})
+		http.ListenAndServe(":8081", nil)
+	}()
+
+	select {}
+}
+
+// startControllersWithLeaderElection uses client-go leader election instead of controller-runtime
+func startControllersWithLeaderElection(ctx context.Context, setupLog logr.Logger, wdsRestConfig, itsRestConfig *rest.Config, wdsName, itsName string, allowedGroupsSet sets.Set[string], ctlrsToStart sets.Set[string], wdsClientMetrics, itsClientMetrics ksmetrics.ClientMetrics) {
+	// Create WDS client for leader election
+	wdsClient, err := kubernetes.NewForConfig(wdsRestConfig)
+	if err != nil {
+		setupLog.Error(err, "unable to create WDS client for leader election")
+		os.Exit(1)
+	}
+
+	// Ensure the namespace exists for leader election
+	namespace := wdsName + "-system"
```

Comment:
Since this namespace is within the WDS, there is no need to include the WDS name in the namespace name. I would go with something like "kubestellar-system".